library/x509_crt.c

```diff
@@ -1435,32 +1571,179 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 }
 #endif /* MBEDTLS_FS_IO */
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ *
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we currently only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108.
+ */
+int mbedtls_x509_get_other_name( const mbedtls_x509_sequence *subject_alt_name,
+                                 mbedtls_x509_san_othername *other_name )
+{
+    int ret;
+    size_t len;
+    unsigned char *p = subject_alt_name->buf.p;
+    const unsigned char *end = p + subject_alt_name->buf.len;
+    mbedtls_x509_buf cur_oid = { 0, 0, NULL };
+
+    if( subject_alt_name->buf.tag != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 0 ) )
+    {
+        /*
+         * The given subject alternative name is not of type "othername".
+         */
+        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+    }
+
+    if( ( ret = mbedtls_asn1_get_tag( &p, end, &len,
+                                      MBEDTLS_ASN1_OID ) ) != 0 )
+        return( MBEDTLS_ERR_X509_INVALID_EXTENSIONS + ret );
+
+    cur_oid.tag = MBEDTLS_ASN1_OID;
+    cur_oid.p = p;
+    cur_oid.len = len;
```

Comment:
Do we really want the user to see the raw OID? Wouldnt it be preferrable to simple give him some constant or something that say "its type X"? Its much easier to compare a contant integer than a string...

Reply:
But:
1. This is only in a temporary variable, to verify the the oid is for hardware module name
2. The `otherName` type is very generic, and can have numerous oid types. We can't foresee all the types there is

Reply:
ok